cns/cnsclient/cnsclient.go

```diff
@@ -14,86 +16,74 @@ import (
 	"github.com/pkg/errors"
 )
 
-// CNSClient specifies a client to connect to Ipam Plugin.
-type CNSClient struct {
-	connectionURL string
-	httpc         http.Client
-}
-
 const (
-	defaultCnsURL   = "http://localhost:10090"
 	contentTypeJSON = "application/json"
+	defaultBaseURL  = "http://localhost:10090"
+	// DefaultTimeout default timeout duration for CNS Client.
+	DefaultTimeout = 5 * time.Second
 )
 
-var cnsClient *CNSClient
-
-// InitCnsClient initializes new cns client and returns the object
-func InitCnsClient(url string, requestTimeout time.Duration) (*CNSClient, error) {
-	if cnsClient == nil {
-		if url == "" {
-			url = defaultCnsURL
-		}
-
-		cnsClient = &CNSClient{
-			connectionURL: url,
-			httpc: http.Client{
-				Timeout: requestTimeout,
-			},
-		}
-	}
-
-	return cnsClient, nil
+// Client specifies a client to connect to Ipam Plugin.
+type Client struct {
+	client  http.Client
+	baseURL url.URL
 }
 
-// GetCnsClient returns the cns client object
-func GetCnsClient() (*CNSClient, error) {
-	var err error
+// New returns a new CNS client configured with the passed URL and timeout.
+func New(baseURL string, requestTimeout time.Duration) (*Client, error) {
+	if baseURL == "" {
+		baseURL = defaultBaseURL
+	}
 
-	if cnsClient == nil {
-		err = &CNSClientError{
-			types.UnexpectedError,
-			//nolint:goerr113
-			fmt.Errorf("[Azure CNSClient] CNS Client not initialized"),
-		}
+	u, err := url.Parse(baseURL)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to parse base URL %s", baseURL)
 	}
 
-	return cnsClient, err
+	return &Client{
+		baseURL: *u,
+		client: http.Client{
+			Timeout: requestTimeout,
+		},
+	}, nil
 }
 
 // GetNetworkConfiguration Request to get network config.
-func (cnsClient *CNSClient) GetNetworkConfiguration(orchestratorContext []byte) (
-	*cns.GetNetworkContainerResponse, error) {
-	var body bytes.Buffer
-
-	url := cnsClient.connectionURL + cns.GetNetworkContainerByOrchestratorContext
-	log.Printf("GetNetworkConfiguration url %v", url)
+func (c *Client) GetNetworkConfiguration(orchestratorContext []byte) (*cns.GetNetworkContainerResponse, error) {
```

Comment:
I see that context has been added to some methods below. Let's take this chance and add them to all client's methods.